ffmpeg/lpms_ffmpeg.c

```diff
@@ -670,6 +875,11 @@ int process_in(struct input_ctx *ictx, AVFrame *frame, AVPacket *pkt)
     else if (pkt->stream_index == ictx->vi || pkt->stream_index == ictx->ai) break;
     else dec_err("Could not find decoder or stream\n");
```

Comment:
Does this means that we're always doing decoding, even for stream copy?

Reply:
Stream copy does not decode if all other outputs for that stream are also copy (or drop). 

https://github.com/livepeer/lpms/blob/8d03c6b7397deaed1a2bf3cb2bafdabe0bcaa79e/ffmpeg/lpms_ffmpeg.c#L232-L236

https://github.com/livepeer/lpms/blob/8d03c6b7397deaed1a2bf3cb2bafdabe0bcaa79e/ffmpeg/lpms_ffmpeg.c#L1279-L1283

(Note there's a little trick here used to force decoding if there's no outputs - we depend on that when checking pixel counts!)

If all the outputs of a given stream are copy (or drop), then we shouldn't decode - the flow will hit line 875 (right above this line) to check the stream index, and exits early from the decode loop.

https://github.com/livepeer/lpms/blob/8d03c6b7397deaed1a2bf3cb2bafdabe0bcaa79e/ffmpeg/lpms_ffmpeg.c#L875

Reply:
Thanks!